app/models/concerns/hyrax/solr_document_behavior.rb

```diff
@@ -91,9 +91,9 @@ def creator
     end
 
     def visibility
```

Comment:
> if this solves the UI problem for #6235, i'd suggest we move the `embargo_history` behavior from `#deactivate!` into `#release` (i think we want this history data anytime an embargo is released), and deprecate that method for removal.

I do think that this may solve the UI problem, but not the initial problem I was trying to solve. which is that running `hyrax:embargo:deactivate_expired` does **not** remove the embargo. in my testing yesterday, I manually ran `#release_embargo_for` on a work then checked the work show page and the embargo was still applied. as well as the work still showing on the "/embargoes" page. that's why the nullifying and saving came into play. nullifying the embargo may not be required, I added it to follow the `deactivate!` pattern. either way, I think the saving of the work/embargo is required. which may be in part to the fact that embargoes are now referenced on a work with an `embargo_id` attribute, and not saved on the work itself. this was done to be in line with how valkyrie handles embargoes. (more about that can be found in the code comment for #6098)

I'm fine with moving the history setting into `#release`. (I considered making that change on #6235 but wasn't sure if it was desired since history wasn't being set at all until #6098). however, I think that deprecating `deactivate!` will lead us to having the same problem of not being able to deactivate an embargo in valkyrie. that's why I created it. (mimicking behavior in hydra-head). I'm open to other ideas, but any changes that are made on that front would need to be tested to ensure that the intended functionality for #5844 and #6126 still work. (I won't have the time today to test this out and I'm OOO the rest of the week)

Reply:
i'm pretty confused. i had understood that the problem you were trying to solve was that releasing the embargo in `EmbargoManager` doesn't remove it from the UI's embargo management table.

it seems to me that the existing `EmbargoManager#release` tests ensure that `#release` *does* in fact end enforcement of the embargo (in that it changes the object visibility to be the correct post-embargo visibility). what i found was that the UI was not checking whether the embargo is actually enforced, just looking for any indexed embargo release date. that doesn't seem adequate, so this proposes patching it.

if there's another problem, i think it would be helpful if we could work on getting a test in that demonstrates the issue.

Reply:
I might be able to look into adding some tests next week, depending on what my schedule looks like when I return. however, https://github.com/scientist-softserv/atla-hyku/issues/121 is original problem I was aiming to solve. 
when I tested the rake tasks against hyrax `main` yesterday, they didn't return 400 errors, but they didn't remove the embargo either. my intention was to fix that problem in hyrax, then move it to atla.

Reply:
>  the work still showed on the "/embargoes" page

this is the main issue this tries to patch. the problem, as i see it, is that *even if* the embargo is released (i.e. the correct post-embargo permissions are applied, the public can access the object, etc...), the `/embargoes` page continues to list any object that has an indexed `#embargo_release_date`. this was a shortcut implementation of the `Hyrax::EmbargoService` dating at least 2014.

i started out by trying to write a test for the `EmbargoesControllerBehavior` that demonstrated the issue, but found that it doesn't actually populate a list of embargoed objects. instead, it renders the view, which calls `EmbargoHelper`, and then the `EmbargoService`. this last had tests that explicitly ensured the behavior above, so i rewrote them to ensure the correct behavior instead: `#assets_under_embargo` should return only assets where the embargo is currently being enforced, not just ones that have an `embargo_release_date` attached.

i think the reason we've been chasing this issue around the code base is that we've not paid attention to this root cause. there may still be other issues, but i think it's necessary to ensure the UI reflects the actual object state before we try to fix the problem by manipulating object state.

----

> and still had the "embargo" badge on the show page and on "/dashboard/my/works"

i think the changes to `SolrDocument` should address this; though admittedly i didn't analyze this as closely. i just noticed that it was using the same shorthand logic that the `EmbargoService` was using, and it was helpful for both fixes to resolve it in `SolrDocument`.

----

if this doesn't get us on the same page, can we jump on a call to chat about it. i'm concerned we're talking past each other.